src/worker/indexers/chainIndexer.ts

```diff
@@ -0,0 +1,285 @@
+import { SmartContract } from "@thirdweb-dev/sdk";
+import { ethers } from "ethers";
+import { getBlockForIndexing } from "../../db/chainIndexers/getChainIndexer";
+import { upsertChainIndexer } from "../../db/chainIndexers/upsertChainIndexer";
+import { prisma } from "../../db/client";
+import {
+  bulkInsertContractEventLogs,
+  type ContractEventLogEntry,
+} from "../../db/contractEventLogs/createContractEventLogs";
+import { getContractSubscriptionsByChainId } from "../../db/contractSubscriptions/getContractSubscriptions";
+import { getConfig } from "../../utils/cache/getConfig";
+import { getContract } from "../../utils/cache/getContract";
+import { getSdk } from "../../utils/cache/getSdk";
+import { logger } from "../../utils/logger";
+
+export interface GetSubscribedContractsLogsParams {
+  chainId: number;
+  contractAddresses: string[];
+  fromBlock: number;
+  toBlock: number;
+}
+
+/**
+ * Used to abstract eth_getLogs
+ * requirement: throw if any of the rpc calls fail, otherwise some contracts will be out of sync
+ * @param params
+ * @returns ethers.Logs[]
+ */
+export const ethGetLogs = async (params: GetSubscribedContractsLogsParams) => {
+  /* this should use log filter: address: [...contractAddresses] when thirdweb supports */
+  const sdk = await getSdk({ chainId: params.chainId });
+  const provider = sdk.getProvider();
+
+  logger({
+    service: "worker",
+    level: "debug",
+    message: `Starting log fetch - ChainIndexer: ${params.chainId}`,
+  });
+
+  const startTime = performance.now();
```

Comment:
we should remove these as we do need this in main. good when developing as we want to see how much time it takes etc, but not required for `main`